lib/Runtime/Library/JSONStringifier.cpp

```diff
@@ -313,9 +311,10 @@ JSONStringifier::ToJSON(_In_ JavascriptString* key, _In_ RecyclableObject* value
 uint32
 JSONStringifier::ReadArrayLength(_In_ RecyclableObject* value)
```

Comment:
Should we have templated version for just an array (common case I believe) ?

Reply:
ReadArrayLength in particular won't get much value (since it's once per array), but ReadArrayElement might. Only thing I'm wondering is do you know if there could be a conversion to ES5Array, and if so would that conversion make it incorrect to call DirectGetItem?

If I need to check if I have a JavascriptArray at each element, then template specialization won't get us anything. Otherwise seems like it could be worth it.

Reply:
For now i'll just open an issue. I made the change but it isn't very clean right now and i want to spend some time making it better